features/370-auth-ui/design.md

```diff
@@ -0,0 +1,531 @@
+  |                                                         |                                              |
+  |                                                         |                                              |
+  |                                                         |                                              |
+  +                                                         +                                              +
+```
+
+This flow allows both Auth UI and Auth Gear to write host-only cookie.
+
+## Domain
+
+Auth UI is available for both default domain and custom domain.
+
+### Default domain
+
+If the app is using default domain, then Auth UI is served at `https://accounts.<appname>.skygearapp.com`. Note that this domain format may conflict with deployment version. Auth UI has higher precedence than deployment version. The choice of `accounts` is borrowed from some other internet services.
+
+### Custom domain
+
+If the app is using custom domain, then Auth UI is served at the developer's wish.
+
+### Within the same domain of the app
+
+Auth UI cannot be served in the same domain with the app. So Auth UI is not available at `https://<appname>.skygearapp.com/authorize` nor `https://www.myapp.com`.
+
+### Domain Example
+
+|   |Default domain|Auth UI default domain|www.myapp.com|accounts.myapp.com|
+|---|---|---|---|---|
+|App|`https://myapp.skygearapp.com`|N/A|`https://www.myapp.com`|N/A|
+|Auth Gear|`https://myapp.skygearapp.com/_auth/`|N/A|`https://www.myapp.com/_auth/`|N/A|
+|Auth UI|N/A|`https://accounts.myapp.skygearapp.com`|N/A|`https://accounts.myapp.com`|
+
+## Logout
+
+Since Auth UI and Auth Gear do not share session, the user logs out with the existing Auth Gear API. Auth UI does **NOT** provide the Logout Endpoint.
+
+OpenID Connect does have 3 draft specifications related to logout. They are
+
+- [Session Management](https://openid.net/specs/openid-connect-session-1_0.html)
+- [Front-Channel Logout](https://openid.net/specs/openid-connect-frontchannel-1_0.html)
+- [Back-Channel Logout](https://openid.net/specs/openid-connect-backchannel-1_0.html)
+
+These specs is useful for implementing Single logout.
+
+## Page templates
+
+All pages served by Auth UI are rendered with Go standard library package [html/template](https://golang.org/pkg/html/template/).
+This is also the package we use to render HTML email template. So the developer only needs to learn the syntax once.
+
+The content of the template is mainly HTML with some CSS.
+It also contains some JavaScript to add some extra functionalities such as dynamically checking password requirements and toggling of password visibility.
+All pages must be function properly even without JavaScript.
+
+### Customization of page templates
+
+The developer can customize the pages at 2 levels.
+They can keep using the builtin templates and customize the theme.
+Or they can replace the builtin ones with their own ones.
+
+#### Customizing the theme with CSS
```

Comment:
What about the logo?

Reply:
Added in https://github.com/SkygearIO/features/pull/411/commits/a3a962547bd82dce0c7a3ee11b1668e80baf208c